src/c-opentimelineio/tests/OTIOCompositionTests.cpp

```diff
@@ -59,6 +59,43 @@ class OTIONestingTest : public ::testing::Test
 protected:
     void SetUp() override {}
     void TearDown() override {}
+
+    struct ClipWrapperPair
+    {
+        Clip*  clip;
+        Stack* wrapper;
+    };
+    typedef struct ClipWrapperPair ClipWrapperPair;
+
+    ClipWrapperPair _nest(Clip* item, int index)
+    {
+        ClipWrapperPair clipWrapperPair;
+        clipWrapperPair.clip    = NULL;
+        clipWrapperPair.wrapper = NULL;
+
+        if(item == NULL) return clipWrapperPair;
+
+        Composition* parent = Composable_parent((Composable*) item);
+
+        if(parent == NULL) { return clipWrapperPair; }
+        OTIOErrorStatus* errorStatus = OTIOErrorStatus_create();
+        Stack*           wrapper = Stack_create(NULL, NULL, NULL, NULL, NULL);
+
+        Clip* clip = (Clip*) SerializableObject_clone(
+            (SerializableObject*) item, errorStatus);
+
+        /* now put the item inside the wrapper */
+        bool appendOK = Composition_append_child(
+            (Composition*) wrapper, (Composable*) clip, errorStatus);
+
+        /* swap out the item for the wrapper */
+        bool setOK = Composition_set_child(
+            (Composition*) parent, index, (Composable*) wrapper, errorStatus);
+
+        clipWrapperPair.clip    = clip;
+        clipWrapperPair.wrapper = wrapper;
+        return clipWrapperPair;
+    }
```

Comment:
@meshula @jminor @ssteinbach @reinecke 

I'm trying to translate this:

https://github.com/PixarAnimationStudios/OpenTimelineIO/blob/eaccea3b61455705286ede0de239dd84406352bb/tests/test_composition.py#L1611

The comment below is according to the flow in the python definition of the function.

The problem I'm facing here is how to swap the `item` for the `wrapper`. I have just the pointer to the clip in the track.
Once I set the child at `index 0` to `wrapper` the `clip` has no references and will get destroyed. So the next thing would be to try to add the `clip` to the `wrapper` before setting the child. But the clip is a child of the track already so it cannot be added to come other composition.

So my query here would be how to get ownership of the clip?

For now I have implemented a workaround that I don't really like. I've cloned the `clip`, put that into the `wrapper` and then set the child in the parent. Then I've returned a struct with the `wrapper` and new `clip` pointer for further nesting.